Sources/SwiftDocC/Catalog Processing/GeneratedCurationWriter.swift

```diff
@@ -0,0 +1,184 @@
+        guard let node = context.documentationCache[reference],
+              let symbol = node.semantic as? Symbol,
+              let automaticTopics = try? AutomaticCuration.topics(for: node, withTraits: [], context: context),
+              !automaticTopics.isEmpty
+        else {
+            return nil
+        }
+        
+        let relativeLinks = linkResolver.disambiguatedRelativeLinksForDescendants(of: reference)
+        
+        // Any reference that is already curated should be excluded from the auto-generated curation
+        let alreadyCurated = Set((symbol.topics?.taskGroups ?? []).flatMap { $0.links.compactMap(\.destination) })
+        
+        // Top-level curation has a few special behaviors regarding symbols with different representations in multiple languages.
+        let isForTopLevelCuration = symbol.kind.identifier == .module
+        
+        var text = ""
+        for taskGroup in automaticTopics {
+            if isForTopLevelCuration, let firstReference = taskGroup.references.first, context.documentationCache[firstReference]?.symbol?.kind.identifier == .typeProperty {
+                // Skip type properties in top-level curation. It's not clear what's the right place for these symbols are since they exist in
+                // different places in different source languages (which documentation extensions don't yet have a way of representing).
+                continue
+            }
+            
+            let links: [(link: String, comment: String?)] = taskGroup.references.compactMap { (curatedReference: ResolvedTopicReference) -> (String, String?)? in
+                guard !alreadyCurated.contains(curatedReference.absoluteString) else { return nil }
+                guard let linkInfo = relativeLinks[curatedReference] else { return nil }
+                // If this link contains disambiguation, include a comment with the full symbol declaration to make it easier to know which symbol it refers to.
+                if linkInfo.hasDisambiguation,
+                   let symbol = context.documentationCache[curatedReference]?.symbol,
+                   let fragments = symbol[mixin: SymbolGraph.Symbol.DeclarationFragments.self]?.declarationFragments
+                {
+                    let commentText = fragments.map(\.spelling).joined().split(whereSeparator: { $0.isWhitespace || $0.isNewline }).joined(separator: " ")
+                    return (linkInfo.link, commentText)
+                }
+                return (linkInfo.link, nil)
+                
+            }
+            
+            guard !links.isEmpty else { continue }
+            
+            text.append("\n\n### \(taskGroup.title ?? "<!-- This auto-generated topic has no title -->")\n")
+            
+            let longestLink = links.map(\.link.count).max()! /* `links` are non-empty */ + 7 /* the extra characters used to make the link "\n- ``" before and "``" after */
```

Comment:
This is a slightly confusing variable name, took a little while to understand this measures the longest line without a comment. I wonder if generating the String components first and then calculating padding might be more readable, even though it introduces an extra loop over the links, something along these lines:
```swift
let linkTextFragments = links.map { (link: "\n- ``\($0.link)``", comment: $0.comment.map { " <!-- \($0) -->" }) }
let longestLinkLine = linkTextFragments.map(\.link.count).max()!
let curationText = linkTextFragments
    .map { $0.comment == nil ? $0.link : $0.link.padding(toLength: longestLinkLine, withPath: " ", startingAt: 0).append($0.comment!) }
    .joined()
text.append(curationText)
```

Reply:
I incorporated the code that wraps the link and comment text in markup into the `compactMap` above. That made the code that computes the longest link and appends the markup strings easier to follow.